lib/Locale/Maketext/Test.pm

```diff
@@ -0,0 +1,565 @@
+the correct number of plural forms as parameters.
+
+=head3 incorrect usage of %d in %plural() parameters
+
+In some languages, like English or German, singular is applicable only to the
+quantity of 1. That means the German translator could come up for instance
+with the following valid %plural call:
+
+    %plural(%5,ein Stein,%d Steine)
+
+In other languages, like French or Russian, this would be an error. French uses
+singular also for 0 quantities. So, if the French translator calls:
+
+    %plural(%5,une porte,%d portes)
+
+and in the actual call the quantity of 0 is passed the output is still "une porte".
+In Russian the problem is even more critical because singular is used for instance
+also for the quantity of 121.
+
+Thus, this test checks if a) the target language is similar to English in having only
+2 plural forms, singular and one plural, and in applying singular only to the quantity
+of 1. If both of these conditions are met %plural calls like the above are allowed.
+Otherwise, if at least one of the parameters passed to %plural contains a %d,
+all of the parameters must contain the %d as well.
+
+That means the following 2 %plural calls are allowed in Russian:
+
+    %plural(%3,%d книга,%d книги,%d книг)
+    %3 %plural(%3,книга,книги,книг)
+
+while this is forbidden:
+
+    %plural(%3,одна книга,%d книги,%d книг)
+
+=head1 SUBROUTINES/METHODS
+
+=head2 debug
+
+set this if you need to check warnings along with errors
+
+=cut
+
+use Moose;
+use namespace::autoclean;
+
+has debug => (
+    is      => 'ro',
+    isa     => 'Bool',
+    default => 0
+);
+
+=head2 directory
+
+directory where locales files are located
+
+=cut
+
+has directory => (
+    is       => 'ro',
+    isa      => 'Str',
+    required => 1
+);
+
+=head2 languages
+
+language array, set this if you want to test specific language only in specified directory
+
+=cut
+
+has languages => (
+    is      => 'rw',
+    isa     => 'ArrayRef[Str]',
+    default => sub { [] });
+
+=head2 auto
+
+flag to fallback when a key is missing from lexicons
+
+    # if this is not set then maketext will output errors if
+    # translations is marked as fuzzy or is missing
+    # read more about it here https://metacpan.org/pod/Locale::Maketext::Lexicon
+    Locale::Maketext::Lexicon->import({ _auto => 1 })
+
+=cut
+
+has auto => (
+    is      => 'ro',
+    isa     => 'Bool',
+    default => 0
+);
+
+has _status => (
+    is       => 'rw',
+    isa      => 'HashRef',
+    lazy     => 1,
+    init_arg => undef,
+    default  => sub {
+        {
+            status   => 1,
+            errors   => {},
+            warnings => {}};
+    });
+
+=head2 BUILD
+
+=cut
+
+sub BUILD {
+    my $self = shift;
+
+    unless (scalar @{$self->languages}) {
+        my @lang = ();
+        if (opendir my $dh, $self->directory) {
+            while (readdir $dh) {
+                if (my ($x) = grep { $_ =~ /^(\w+)\.po$/ } $_) {
```

Comment:
or even:
```perl
push @lang, /^(\w+)\.po$/;
```
since that's an empty list without a match (that's what makes the `if` clause work in the first place). If you find the explicit `if()` more readable I'm fine with that.